server/backend/database/version_vector.go

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2024 The Yorkie Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package database
+
+import (
+	"github.com/yorkie-team/yorkie/api/types"
+	"github.com/yorkie-team/yorkie/pkg/document/time"
+)
+
+// VersionVectorInfo is a structure representing information about the version vector for each document and client.
+type VersionVectorInfo struct {
+	ID            types.ID           `bson:"_id"`
+	ProjectID     types.ID           `bson:"project_id"`
+	DocID         types.ID           `bson:"doc_id"`
+	ClientID      types.ID           `bson:"client_id"`
+	VersionVector time.VersionVector `bson:"version_vector"`
+}
```

Comment:
Is there an advantage to storing `VersionVector` per `Client` rather than the min value of VersionVector per `Document`?

Reply:
Storing `VersionVector` its not an option, it's inevitable.

I'll explain this when we have meeting

Reply:
@JOOHOJANG wrote and shared the exception situation below.

<img width="658" alt="Screenshot 2024-09-23 at 5 52 14 PM" src="https://github.com/user-attachments/assets/34e227e8-37db-4283-9dfd-b7a460b46a4b">

The problem with storing only the min value of VersionVector on the server is that we can properly calculate the min value for other peer's VV, but when calculating the min value for own previous version vector on the server, we cannot update it to the next version vector.

e.g.) 
- min(A, B, C, ....): ✅
- min(prevA, nextA) => prevA: ❌

Reply:
https://discord.com/channels/928301813785038878/1263454638254985226/1284163377320427550